tuftool/src/refresh.rs

```diff
@@ -0,0 +1,197 @@
+    #[structopt(short = "j", long = "jobs")]
+    jobs: Option<NonZeroUsize>,
+
+    /// Key files to sign with
+    #[structopt(short = "k", long = "key", required = true)]
+    keys: Vec<KeySource>,
+
+    /// Version of snapshot.json file
+    #[structopt(long = "snapshot-version")]
+    snapshot_version: NonZeroU64,
+    /// Expiration of snapshot.json file; can be in full RFC 3339 format, or something like 'in
+    /// 7 days'
+    #[structopt(long = "snapshot-expires", parse(try_from_str = parse_datetime))]
+    snapshot_expires: DateTime<Utc>,
+
+    /// Version of targets.json file
+    #[structopt(long = "targets-version")]
+    targets_version: NonZeroU64,
+    /// Expiration of targets.json file; can be in full RFC 3339 format, or something like 'in
+    /// 7 days'
+    #[structopt(long = "targets-expires", parse(try_from_str = parse_datetime))]
+    targets_expires: DateTime<Utc>,
+
+    /// Version of timestamp.json file
+    #[structopt(long = "timestamp-version")]
+    timestamp_version: NonZeroU64,
+    /// Expiration of timestamp.json file; can be in full RFC 3339 format, or something like 'in
+    /// 7 days'
+    #[structopt(long = "timestamp-expires", parse(try_from_str = parse_datetime))]
+    timestamp_expires: DateTime<Utc>,
+}
+
+impl RefreshArgs {
+    pub(crate) fn run(&self) -> Result<()> {
+        if let Some(jobs) = self.jobs {
+            rayon::ThreadPoolBuilder::new()
+                .num_threads(usize::from(jobs))
+                .build_global()
+                .context(error::InitializeThreadPool)?;
+        }
+
+        let settings = tough::Settings {
+            root: File::open(&self.root).unwrap(),
+            datastore: self.workdir.as_path(),
+            metadata_base_url: self.metadata_base_url.as_str(),
+            target_base_url: self.metadata_base_url.as_str(),
+            limits: Limits::default(),
+        };
+
+        if self.metadata_base_url.scheme() == "file" {
+            let repository =
+                Repository::load(&FilesystemTransport, settings).context(error::Metadata)?;
+            self.refresh(&repository)
+        } else {
+            let transport = HttpTransport::new();
+            let repository = Repository::load(&transport, settings).context(error::Metadata)?;
+            self.refresh(&repository)
+        }
+    }
+
+    fn refresh<'a, T: Transport>(&self, repository: &Repository<'a, T>) -> Result<()> {
+        // clone the targets.json file but with a new version number and expiration date
+        let new_targets = Targets {
+            spec_version: crate::SPEC_VERSION.to_owned(),
+            version: self.targets_version,
+            expires: self.targets_expires,
+            targets: repository.targets().signed.targets.clone(),
+            _extra: HashMap::new(),
+            // TODO - get help. Clippy wants _extra renamed to extra if we do this. Should we do so?
+            // _extra: repository.targets().signed._extra.clone(),
```

Comment:
I need some guidance here. If I 'bind' _extra by cloning the loaded _extra, then clippy wants it rename to remove the underscore. Not sure I can avoid this by mutating the loaded object? Or should we do the rename?

Reply:
I have just seen that #20 calls _extra into question.

Reply:
Per offline convo, added `#![allow(clippy::used_underscore_binding)]` to main.rs